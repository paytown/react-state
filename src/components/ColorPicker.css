.ColorPicker {
  button {
    border: none;
    padding: 4em;
    outline: none;
    margin: 1em;
  }
  
  .red {
    color: white;
    background-color: red;
  }
  
  .green {
    color: white;
    background-color: green;
  }
  
  .blue {
    color: white;
    background-color: blue;
  }
}
