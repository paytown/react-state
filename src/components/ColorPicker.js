import React from 'react';
import PropTypes from 'prop-types';
import style from './ColorPicker.css';

function ColorPicker({ selectColorHandler }) {
  return (
    <section className={style.ColorPicker}>
      <button className={style.red} onClick={() => selectColorHandler('red')}>red</button>
      <button className={style.green} onClick={() => selectColorHandler('green')}>green</button>
      <button className={style.blue} onClick={() => selectColorHandler('blue')}>blue</button>
    </section>
  );
}

ColorPicker.propTypes = {
  selectColorHandler: PropTypes.func.isRequired
};

export default ColorPicker;
