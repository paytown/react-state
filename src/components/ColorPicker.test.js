import React from 'react';
import { shallow } from 'enzyme';
import ColorPicker from './ColorPicker';

describe('Color Picker buttons', () => {
  it('renders clickable buttons', () => {
    const wrapper = shallow(<ColorPicker />);
    expect(wrapper).toMatchSnapshot();
  });
});
