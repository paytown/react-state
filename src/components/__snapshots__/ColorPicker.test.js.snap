// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Picker buttons renders clickable buttons 1`] = `
<section
  className="ColorPicker"
>
  <button
    className="red"
    onClick={[Function]}
  >
    red
  </button>
  <button
    className="green"
    onClick={[Function]}
  >
    green
  </button>
  <button
    className="blue"
    onClick={[Function]}
  >
    blue
  </button>
</section>
`;
